service/app/controllers/usermanagement/validator/ShadowUserMigrateReqValidator.java

```diff
@@ -26,11 +33,30 @@ private ShadowUserMigrateReqValidator(Request request,String tokenUserId) {
     public static  ShadowUserMigrateReqValidator getInstance(Request request,String callerId){
         return new ShadowUserMigrateReqValidator(request,callerId);
     }
+
     public void validate(){
         checkMandatoryFieldsPresent(request.getRequest(),mandatoryParamsList);
         String userId=(String)request.getRequest().get(JsonKey.USER_ID);
         if(!StringUtils.equalsIgnoreCase(userId,callerId)){
             throw new ProjectCommonException(ResponseCode.invalidParameterValue.getErrorCode(), MessageFormat.format(ResponseCode.invalidParameterValue.getErrorMessage(), userId, JsonKey.USER_ID), ResponseCode.CLIENT_ERROR.getResponseCode());
         }
+        validateAction((String)request.getRequest().get(JsonKey.ACTION));
+    }
+
+    private void validateAction(String action){
+        if(!allowedActions.contains(action)){
+            throw new ProjectCommonException(ResponseCode.invalidParameterValue.getErrorCode(), MessageFormat.format(ResponseCode.invalidParameterValue.getErrorMessage(), action, JsonKey.ACTION+" supported actions are:"+allowedActions), ResponseCode.CLIENT_ERROR.getResponseCode());
+        }
     }
+
+
+    private static List<String> getEnumsAsString(){
+        List<ProjectUtil.MigrateAction> enums=Arrays.asList(ProjectUtil.MigrateAction.values());
+        List<String>actions=new ArrayList<>();
```

Comment:
FILE from where we are picking MigrateActions
https://github.com/project-sunbird/sunbird-utils/pull/740/files